pilot/pkg/config/kube/crd/controller.go

```diff
@@ -58,25 +62,41 @@ var (
 		Help: "Errors converting k8s CRDs",
 	}, []string{"name"})
 
+	k8sResourceVersions = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "pilot_k8s_resource_versions",
+		Help: "Applied CRD resource versions",
+	}, []string{"gvk", "name", "namespace", "resourceVersion", "instanceID"})
```

Comment:
nit: I'm slightly concerned about having a metric named `...resource_versions` and a label of `resourceVersion`. Should we call it `resources` and the label `version` ?

do we want to call `instanceID` something like `discovery_uid` ?